compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -4242,7 +4253,7 @@ object Types {
             // final val one = 1
             // type Two = one.type + one.type
             // ```
-            case tp: TermRef => tp.underlying
+            case tp: TypeProxy if tp.underlying.isStable => tp.underlying.fixForEvaluation
```

Comment:
Reasoning here: no need to check that `tp` is stable because it will be if its underlying type is stable. Could someone confirm if this is correct?

Otherwise, we might want:

```scala
case tp: TypeProxy if tp.isStable && tp.underlying.isStable => tp.underlying.fixForEvaluation
```

or

```scala
case tp: SingletonType if tp.isStable && tp.underlying.isStable => tp.underlying.fixForEvaluation
```

Reply:
I think at least morally this is correct.